### DISCUSSION TITLE
Image preloading in Jatos and JsPych

### DISCUSSION THREAD

**Discussion by Le-Gris** [jspsych/jsPsych#2685] [General]
Hi,

I setup an experiment using JATOS and jsPsych running on an AWS EC2 instance and I'm having issues with image preloading. Looking at the Network tab in the developer tools on Chrome while running the experiment, it seems that although I included a preloading trial for each same-different block in the experiment, the images are being loaded as the trials are being shown. There is a visible lag in the image presentation. 

At first, I thought this was an issue with JATOS and how it handles caching (I am not very familiar with browser headers and caching...) so I added the following to my Apache2 config file `Header set Cache-Control "max-age=3600, public"`. This did not change anything. Now I'm thinking the issue might be on the side of jsPsych... Any help would be appreciated!

EDIT: I am not using auto_preload but providing an array with file paths of the images for preloading. I set the parameter `message: '<h3>Loading images...</h3>'` and it actually appears when I'm running the experiment. But I'm not sure what to look for in the Network tab.

Thanks,
Solim

**Comment by jodeleeuw** [comment]
Hi @AlephG

Can you share a sample of the code you are using for preloading? 

Do you see any network activity when the preload trials are running?

**Reply by Le-Gris** [reply]
I looked at other places where I used preloading and compared network activity. The problem was a typo in my code ... Problem solved. Thanks anyways!
